apierr/errors.go

```diff
@@ -168,6 +168,11 @@ func GetAPIError(ctx context.Context, resp common.ResponseWrapper) error {
 }
 
 func parseErrorFromResponse(resp *http.Response, requestBody, responseBody []byte) *APIError {
+	if len(responseBody) == 0 {
```

Comment:
> This leads to some race conditions because workspace deletion is asynchronous, so attempting to delete a credential while a workspace is running will ultimately fail. 

Just to confirm, with this change the credential deletion will still fail (as workspace is in use) so it doesn't prevent the race condition however we return before unmarshalling empty response body right?

Reply:
Correct. I'm going to make another follow-up PR to address the race condition.